climpred/preprocessing/mpi.py

```diff
@@ -0,0 +1,118 @@
+import glob as _glob
+import os as _os
+
+import numpy as np
+import xarray as xr
+
+
+def get_path(
+    dir_base_experiment='/work/bm1124/m300086/CMIP6/experiments',
+    member=1,
+    init=1960,
+    model='hamocc',
+    output_stream='monitoring_ym',
+    timestr='*1231',
+    ending='nc',
+):
+    """Get the path of a file of for MPI-ESM standard output file names and directory.
+
+    Args:
+        dir_base_experiment (str): Path of experiments folder. Defaults to
+            "/work/bm1124/m300086/CMIP6/experiments".
+        member (int): `member`. Defaults to 1.
+        init (init): `init`. Defaults to 1960.
+        model (str): submodel name. Defaults to "hamocc".
+            Allowed: ['echam6', 'jsbach', 'mpiom', 'hamocc'].
+        output_stream (str): output_stream name. Defaults to "monitoring_ym".
+            Allowed: ['data_2d_mm', 'data_3d_ym', 'BOT_mm', ...]
+        timestr (str): timestr likely including *. Defaults to "*1231".
+        ending (str): ending indicating file format. Defaults to "nc".
+            Allowed: ['nc', 'grb'].
+
+    Returns:
+        str: path of requested file(s)
+
+    """
+    # get experiment_id
+    dirs = _os.listdir(dir_base_experiment)
+    experiment_id = [
+        x for x in dirs if (f'{init}' in x and 'r' + str(member) + 'i' in x)
+    ]
+    assert len(experiment_id) == 1
+    experiment_id = experiment_id[0]
+    dir_outdata = f'{dir_base_experiment}/{experiment_id}/outdata/{model}'
+    path = f'{dir_outdata}/{experiment_id}_{model}_{output_stream}_{timestr}.{ending}'
+    if _os.path.exists(_glob.glob(path)[0]):
+        return path
+    else:
+        raise ValueError(f'Path not found or no access: {path}')
+
+
+def climpred_preprocess_internal(ds, lead_offset=1, time_dim='time'):
+    """CMIP6 DCPP preprocessing before the aggreatations of intake-esm happen."""
+    # set time_dim to integers starting at lead_offset
+    ds[time_dim] = np.arange(lead_offset, lead_offset + ds[time_dim].size)
+    return ds
+
+
+def load_hindcast(
+    inits=range(1961, 1965),
+    members=range(1, 3),
+    preprocess=None,
+    lead_offset=1,
+    parallel=True,
+    engine=None,
+    **get_path_kwargs,
+):
+    """Load multi-member, multi-initialization hindcast experiment into one
+    `xr.Dataset` compatible with `climpred`.
```

Comment:
Make sure any instances of ` ` in docstrings is actually `` ``, since that's what `.rst`/`sphinx` uses to render code.